File: frontend/src/app/dto/catan/GameConfig.ts
import {DiceConfig} from "./DiceConfig";
import {Team} from "../global/Team";
import {GameMode} from "./GameMode";

export interface GameConfig {
    teams: Team[],
    gameMode: GameMode,
    classicDice: DiceConfig,
    eventDice: DiceConfig,
    maxShipTurns: number
}

File: backend/src/main/java/net/zorphy/backend/catan/dto/GameConfig.java
package net.zorphy.backend.catan.dto;

import net.zorphy.backend.main.dto.TeamDetails;

import java.util.List;

public record GameConfig(
        List<TeamDetails> teams,
        GameMode gameMode,
        DiceConfig classicDice,
        DiceConfig eventDice,
        int maxShipTurns
) {

}
